### ISSUE TITLE
[BUG]Get-PnPGroupMember returning user not in group

### ISSUE THREAD

**Issue by JaeEmAre67** [pnp/powershell#1227]
### Notice
Many bugs reported are actually related to the PnP Framework which is used behind the scenes. Consider carefully where to report an issue:

1. **Are you using ```Invoke-PnPSiteTemplate``` or ```Get-PnPSiteTemplate```**? **No** The issue is most likely related to the Provisioning Engine. The Provisioning engine is _not_ located in the PowerShell repo. Please report the issue here: https://github.com/pnp/pnpframework/issues.
1. **Is the issue related to the cmdlet itself, its parameters, the syntax, or do you suspect it is the code of the cmdlet that is causing the issue?** **Yes** Then please continue reporting the issue in this repo.
1. **If you think that the functionality might be related to the underlying libraries that the cmdlet is calling** (We realize that might be difficult to determine), please first double check the code of the cmdlet, which can be found here: https://github.com/pnp/powershell/tree/master/src/Commands. If related to the cmdlet, continue reporting the issue here, otherwise report the issue at https://github.com/pnp/pnpframework/issues

### Reporting an Issue or Missing Feature
Please confirm what it is that your reporting
**Get-PnPGroupMember cmdlet is returning a user object of user(s) not in given group**

### Expected behavior 
Please describe what output you expect to see from the PnP PowerShell Cmdlets
**Will return a user if the user specific is a member of the specified group.**

### Actual behavior 
Please describe what you see instead. Please provide samples of output or screenshots.
**Returning user even though not in specified group.**

### Steps to reproduce behavior
Please include complete script or code samples in-line or linked from [gists](https://gist.github.com/)
```
Connect-PnPOnline https://our.sharepoint.com/site/collection;
$user = Read-Host "Input user Email";
$allGroups = Get-PnPGroup;
foreach($group in $allGroups) {
   Get-PnPGroupMember -Group $group.Title <<or $group.Id>> -User $user
}
```
$allGroups is the list of the two groups in our site collection. $user is in the second one and not the first one. We expect nothing for the first $group and yet we're getting the user for both the first and second groups.

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)
1.7.0

### Which operating system/environment are you running PnP PowerShell on?
- [X] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by KoenZomers** [comment]
Diving into it.

**Comment by KoenZomers** [comment]
Fixed the issue in PR #1236 . Wait for it to be merged and try again using the latest prerelease the day thereafter. Assuming this will fix it, closing this issue. If it turns out not to be fixed for you, feel free to report back and we will reopen this issue.

Thanks for reporting it @JaeEmAre67 !
